### PR TITLE
build(deps): update go to 1.22

### PR DESCRIPTION
**Pull request by melekes**
---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)

### CODE REVIEW

**Comment by cason** [comment]
Can you check the `go.sum`?

```
$ git status 
On branch update-go-version
Your branch is up to date with 'origin/update-go-version'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   go.sum

no changes added to commit (use "git add" and/or "git commit -a")
$ go1.22.7 mod tidy
$ $ git diff --stat
 go.sum | 16 ----------------
 1 file changed, 16 deletions(-)
```

File: go.sum
Comment (cason) [inline go.sum:1]: My local version differs.

**Review by cason — APPROVED** [review]
See my comment below/above, then we are done.

File: tools/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ RUN apt update \
        libleveldb-dev libleveldb1d
 
 FROM build AS install
-ARG ROCKSDB=9.2.1
+ARG ROCKSDB=9.2.3
```
Comment (alesforz) [inline tools/Dockerfile:21]: I think bumping the version of `ROCKSDB` is causing the tests to fail. Is it necessary?

**Pull request merged**